site-src/geps/gep-709.md

```diff
@@ -0,0 +1,260 @@
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of ReferencePolicy.
+    Spec ReferencePolicySpec `json:"spec,omitempty"`
+}
+
+
+// ReferencePolicySpec identifies a cross namespace relationship that is trusted
+// for Gateway API.
+type ReferencePolicySpec struct {
+    // From describes the trusted namespaces and kinds that can reference the
+    // resources described in "To".
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinItems=1
+    From []ReferencePolicyFrom `json:"from"`
+
+    // To describes the resources that may be referenced by the resources
+    // described in "From".
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinItems=1
+    To []ReferencePolicyTo `json:"to"`
+}
+
+// ReferencePolicyFrom describes trusted namespaces and kinds.
+type ReferencePolicyFrom struct {
+    // Group is the group of the referrent.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Group string `json:"group"`
+
+    // Kind is kind of the referrent.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Kind string `json:"kind"`
+
+    // Namespace is the namespace of the referrent.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Namespace string `json:"namespace,omitempty"`
+}
+
+// ReferencePolicyTo describes trusted kinds.
+type ReferencePolicyTo struct {
+    // Group is the group of the referrent.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Group string `json:"group"`
+
+    // Kind is kind of the referrent.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Kind string `json:"kind"`
+}
+```
+
+
+### Benefits
+
+* Conceptually similar to NetworkPolicy.
+* A separate resource enables admins to restrict who can allow cross namespace
+  references.
+* Provides consistent way to control references to any resource from a Route.
+* Can be extended in the future for additional use cases.
+* A single ReferencePolicy resource can be used for a namespace in place of
+  separate handshake config on each Service or Route resource.
+
+#### Exceptions
+If traffic is originating from the same location as the configured Route (such
+as sidecars in some mesh implementations) implementations may choose to ignore
+ReferencePolicy. This should only be done if:
+* Other mechanisms like NetworkPolicy can be used to effectively limit
+  cross-namespace references.
+* The implementation clearly documents that ReferencePolicy is not honored.
+
+This exception is very unlikely to apply to any ingress implementations of the
```

Comment:
Generally agree, but this could be plausible in a single-tenant ingress. For example `ingress` namespace has all Gateways and Routes for the ingress. Some routes reference other namespace. There is no security risk here.

Reply:
How is there no security risk in that scenario? Would it be possible the implementation to know that it was always going to be deployed in that way?

Reply:
After discussing this a bit more, I think there are some instances where ingress implementations deployed in this way could potentially deserve an exception. Fundamentally the distinction is if the implementation would be subject to some other cross-namespace restrictions such as NetworkPolicy. I'll update this section to be a bit more clear.

Reply:
I suggested a wording update here, I think that if we make that change, it covers this usecase already.

Reply:
Completely reworded this section, PTAL.